packages/venia-ui/lib/components/SearchPage/searchPage.js

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import { shape, string } from 'prop-types';
+import { useSearchPage } from '@magento/peregrine/lib/talons/SearchPage/useSearchPage';
+
+import { mergeClasses } from '../../classify';
+import Gallery from '../Gallery';
+import FilterModal from '../FilterModal';
+import { fullPageLoadingIndicator } from '../LoadingIndicator';
+import CategoryFilters from './categoryFilters';
+import defaultClasses from './searchPage.css';
+import PRODUCT_SEARCH from '../../queries/productSearch.graphql';
+
+const SearchPage = props => {
+    const classes = mergeClasses(defaultClasses, props.classes);
+
+    const talonProps = useSearchPage({
+        query: PRODUCT_SEARCH
+    });
+    const {
+        loading,
+        error,
+        data,
+        executeSearch,
+        categoryId,
+        openDrawer,
+        history,
+        location
+    } = talonProps;
+
+    if (loading) return fullPageLoadingIndicator;
+    if (error) {
+        return (
+            <div className={classes.noResult}>
+                No results found. The search term may be missing or invalid.
+            </div>
+        );
+    }
+
+    const { products } = data;
+    const { filters, total_count, items } = products;
+
+    if (items.length === 0) {
+        return <div className={classes.noResult}>No results found!</div>;
+    }
+
+    const maybeCategoryFilters = categoryId ? (
+        <CategoryFilters
+            categoryId={categoryId}
+            executeSearch={executeSearch}
+            history={history}
```

Comment:
The talon for `CategoryFilters` should be the one importing/using `useHistory`. Same with `location` if possible.

Reply:
👍 Should have mentioned I was waiting to refactor `categoryFilters` until this early feedback round, but I agree with you. Safe to assume that would live in a `CategoryFilter` talon, correct?

Reply:
Cool. I agree with @rugh, but I also think this is fine for now. Your call, @tjwiebell.

Reply:
Moved both to `CategoryFilter`.